pytorch_lightning/utilities/enums.py

```diff
@@ -188,3 +231,41 @@ def get_max_depth(mode: str) -> int:
     @staticmethod
     def supported_types() -> List[str]:
         return [x.value for x in ModelSummaryMode]
+
+
+class StrategyType(LightningEnum):
```

Comment:
Can we take advantage of this depreaction and make this protected?

```suggestion
class _StrategyType(LightningEnum):
```

`DistributedType` was never meant to be public. Users should always compare with the string value.

Reply:
we have it publically available inside dqn example

Reply:
yeah, there could be cases of users using it

Reply:
> we have it publically available inside dqn example

That's not a good argument as it already accesses protected variables

`if self.trainer._distrib_type in {DistributedType.DP, DistributedType.DDP2}:`

users having to import and use enums is a source of boilerplate, it also means we are less limited in the changes we can do.

Users should always compare and use their string arguments, in that case

`in {"dp", "ddp"}`, as that's why the `LightningEnum` was introduced.

This is the case for the `stage` argument of `setup`

> there could be cases of users using it

This class is new, nobody is using it as you just wrote it, they will get the deprecation warning for the old one.

Reply:
@carmocca Updated!